frontend/src/components/common/ProfileSidebar.vue

```diff
@@ -0,0 +1,183 @@
+<script setup lang="ts">
+import authService from '@/services/AuthService';
+import { useProfileSidebarVisible } from '../../store/useProfileVisibleStore';
+import userProfileInfo from '../../static/userProfileInfo.json';
+
+// svg icon
+import CloseIcon from '../icons/CloseIcon.vue';
+import LogoutIcon from '../icons/LogoutIcon.vue';
+</script>
+
+<template>
+    <div
+        :class="
+            useProfileSidebarVisible.isProfileVisible ? 'fade-in' : 'fade-out'
+        "
+        @click="useProfileSidebarVisible.toggleVisible()"
+    ></div>
+    <Transition name="slide">
+        <div
+            class="profile-container"
+            v-if="useProfileSidebarVisible.isProfileVisible"
+        >
+            <div class="profile-header">
+                <h2>Profile</h2>
+                <button
+                    class="btn-icon"
+                    @click="useProfileSidebarVisible.toggleVisible()"
+                    aria-label="Close"
+                >
+                    <CloseIcon />
+                </button>
+            </div>
+            <div class="sidebar-body">
+                <div class="img-wrapper">
+                    <img
+                        src="../../assets/images/tyrannosaurus-rex1.png"
+                        alt="User avatar"
+                    />
+                </div>
+                <div class="profile-info">
+                    <p class="profile-name">
+                        {{ userProfileInfo[0].userInfo.name }}
```

Comment:
User can be only one login user at a time I assume, so not sure why "userProfileInfo" is an array.